auth0/resource_auth0_global_client.go

```diff
@@ -0,0 +1,43 @@
+package auth0
+
+import (
+	"errors"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"gopkg.in/auth0.v3"
+	"gopkg.in/auth0.v3/management"
+	"net/url"
+)
+
+func newGlobalClient() *schema.Resource {
+	client := newClient()
+	client.Create = createGlobalClient
+	client.Delete = deleteGlobalClient
+
+	name := client.Schema["name"]
+	name.Required = false
+	name.Computed = true
+
+	return client
+}
+
+func createGlobalClient(d *schema.ResourceData, m interface{}) error {
+	api := m.(*management.Management)
+	clients, err := api.Client.List(withIsGlobal(), management.WithFields("client_id"))
```

Comment:
We can avoid the `withIsGlobal` declaration if we use `management.Parameter("is_global", "true")` inline here.

Reply:
I split this in to two separate functions so if we ever want to create a data source in the future, we can reuse the code.